File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int n = nums.length;

        if (n <= 2) {
            return n;
        }
        int first = nums[0];
        int second = nums[1];
        int i = 2;
        
        for (int j = 2; j < n; j++) {
            if (first != second || nums[j] != second) {
                nums[i] = nums[j];
                first = nums[i - 1];
                second = nums[i];
                i++;
            }
        }
        return i;
    }
}

File: 0080-remove-duplicates-from-sorted-array-ii/0080-remove-duplicates-from-sorted-array-ii.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # if len(nums) <= 2:
        #     return len(nums)

        left = 1

        for right in range(2, len(nums)):
            if not (nums[left - 1] == nums[left] and nums[left] == nums[right]):
                left += 1
                nums[left] = nums[right]
        
        return left + 1